src/Command/BaseCommand.php

```diff
@@ -0,0 +1,57 @@
+<?php
+namespace App\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+
+class BaseCommand extends Command
+{
+    protected function checkRequiredOptionsAreNotEmpty(InputInterface $input): void
+    {
+        $options = $this->getDefinition()->getOptions();
+        foreach ($options as $option) {
+            $name  = $option->getName();
+            $value = $input->getOption($name);
+            if ($option->isValueRequired() && (null == $value || '' == $value)) {
+                throw new \InvalidArgumentException(sprintf('The required option %s is not set', $name));
+            }
+        }
+    }
+
+    protected function configure()
+    {
+        $this
+        ->addArgument('username', InputArgument::REQUIRED, "Username for authentication")
+        ->addArgument('password', InputArgument::REQUIRED, "Password for authentication")
+        ->addOption('apiUrl', null, InputOption::VALUE_OPTIONAL, "Url of the api", "http://127.0.0.1");
+        return $this;
+    }
```

Comment:
Please don't modify parent class method signature. Command::configure returns void .

I don't like this "fluent interface" method call chaining, but if you use it, at least use proper indentation as is shown in Symfony docs:
https://symfony.com/doc/4.4/console.html#configuring-the-command

Note the object line, method call indentation and final ; position

Reply:
Ok, but then I cannot call to parent configure function for common arguments. Another way would be to add such arguments on the definition of the application: https://gist.github.com/dhrrgn/8847309 
But then the arguments would get defined outside the command classes. Or we could just repeat the code in each command.

Reply:
I don't understand. You're modifying an object state:

parent::configure();
$this->addWhatever();

should work, did you try?

Reply:
Yes, it says it is not compatible:

> PHP Fatal error:  Declaration of App\Command\DeleteClientCommand::configure() must be compatible with App\Command\BaseCommand::configure(): void in /home/uanabitarte/elkarbackup_api_cli/src/Command/DeleteClientCommand.php on line 9